src/permission.js

```diff
@@ -165,13 +165,16 @@ export function permissionsToFormData(permissionsObject: Object): Object {
  * Transform the permissions form result into data processable by the API.
  * See `permissionsToFormData()` and `preparePermissionsForm()`.
  */
-export function formDataToPermissions(formData: Object) : Object {
-  const {anonymous, authenticated, principals} = formData;
-  const permissionsList = [
-    ...principals,
+export function formDataToPermissions(bid: string, formData: Object) : Object {
+  const {anonymous, authenticated, groups, principals} = formData;
+  const fromGeneric = [
     {principal: EVERYONE, permissions: anonymous},
     {principal: AUTHENTICATED, permissions: authenticated}];
```

Comment:
Nit: please move the closing square bracket to a new line (general coding style)